File: 1199-minimum-time-to-build-blocks/1199-minimum-time-to-build-blocks.py
class Solution(object):
    def minBuildTime(self, blocks, split):
        """
        :type blocks: List[int]
        :type split: int
        :rtype: int
        """
        heapq.heapify(blocks)
        while len(blocks) > 1:
            a, b = heapq.heappop(blocks), heapq.heappop(blocks)
            heapq.heappush(blocks, b + split)
        
        return heapq.heappop(blocks)

File: 1199-minimum-time-to-build-blocks/1199-minimum-time-to-build-blocks.java
class Solution {
    public int minBuildTime(int[] blocks, int split) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for(int b : blocks) pq.add(b);
        while(pq.size() > 1) {
            int a = pq.poll(), b = pq.poll();
            pq.add(split + b);
        }
        return pq.poll();
    }
}